requirements/docs.txt

```diff
@@ -1,9 +1,6 @@
-# On Nov 11, 2021, Sphinx 4.3.0 broke the documentation
-# generation. Remove the < pin when we get a better understanding
-# of why
-# https://github.com/scikit-image/scikit-image/issues/6030
-sphinx>=1.8,<4.3
-sphinx-gallery>=0.7.0,!=0.8.0
+sphinx>=1.8
+# sphinx-gallery 0.10.0 broke support for Sphinx >=4.3.0
+sphinx-gallery>=0.7.0,!=0.8.0,!=0.10.0
```

Comment:
@hmaarrfk I don't think this is correct: I think *any* version of sphinx-gallery prior to 0.10.1 will break with older sphinx. ie we need:

sphinx>=1.8,<4.3
sphinx-gallery

OR

sphinx>=1.8
sphinx-gallery>=0.10.1

I don't think pip lets one specify such complex dependencies, so I think we should just bite the bullet and depend on sphinx-gallery 0.10.1+.

Reply:
hmmm. ok. let's see what the latest builds say. i guess I'm building 0.9 and 4.3 now and we can check the compatibility issues.

otherwise, i agree that we should depend on 0.10.1+

Reply:
Seems like you are right. I added a comment in case downstream wants to rebuild docs (not sure who does that) but let me know if there is anything else.